scripts/test_dsconverter.py

```diff
@@ -0,0 +1,432 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Third Party
+# Import the script code
+from dsconverter import CustomDocstringConverter
+import pytest
+
+
+# Test case for convert_to_google_style function
+def test_get_docstrings():
+    converter = CustomDocstringConverter()
+    test_path = "/Users/mwjohnson/Code/caikit/caikit/core/modules/config.py"
+    docstrings = converter.extract_docstrings_from_file(test_path)
+    for docstring in docstrings:
+        print("foo:\n", docstring)
+    assert 1 == 2
+
+
+def test_basic_custom_docstring():
+    converter = CustomDocstringConverter()
+    test1 = '''
+    """Method for extracting pred set from dataset. Implemented in subclass.
+
+        Args:
+            dataset:  object
+                In-memory version of whatever
+            preprocess_func:  method
+                Function used as proxy for any preliminary steps that need to be taken to run the
+                model on the input text. This helper function ultimately leads to the input to this
+                module and may involve executing other modules.
+            foo
+                Description
+            *args, **kwargs: dict
+                Optional keyword arguments for prediction set extraction.
+        Returns:
+            list
+                List of labels in the format of the module_type that is being called.
+        """
+    '''
+
+    converted = converter.convert_to_google_style(test1)
+    print("Pre-conversion:\n", test1, "\nPost-conversion:\n", converted)
+
+    assert 1 == 2
```

Comment:
So I like that there are "tests" here, though it looks like they all just fail and print out the converted strings so that a human can look through and approve them. I don't think that's quite at the level of polish that we would want to include in caikit (though maybe it's not the end of the world since this isn't in the main unit test suite).

This style of testing could be easily handled with data-driven testing, if you wanted to dig into that a bit. Basically you don't have to write tests for each of these cases, you can simply store some gabe-style docstrings and their expected conversions, and write one test that makes sure the conversions are all correct.

Reply:
Good point. I have added functionality to the test cases so they actually work as tests now.